src/net.cpp

```diff
@@ -671,6 +671,231 @@ void SocketSendData(CNode *pnode)
 
 static list<CNode*> vNodesDisconnected;
 
+class CNodeRef {
```

Comment:
Ugh, is this just a refcounting wrapper around a ``CNode*``? I know it's better to follow upstream exactly here, but if we needed anything like this, I'd like to think that we'd just use ``shared_ptr``.